src/time_bucket.c

```diff
@@ -415,7 +415,12 @@ ts_time_bucket_ng_date(PG_FUNCTION_ARGS)
 	{
 		ereport(ERROR,
 				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
-				 errmsg("origin must be the first day of the month")));
+				 errmsg("origin \"%04d-%02d-%02d\" must be the first day of the month",
+						origin_year,
+						origin_month,
+						origin_day),
+				 errhint("when using timestamptz-version of time_bucket_ng(), 'origin' is "
+						 "converted to provided 'timezone'")));
```

Comment:
A hint should start capitalized and end with a period. See PostgreSQL error style guide. There it might be worth also looking at how and when to use quotes. I would also avoid using the function name in the message as it is self-evident which function this concerns.

```suggestion
				 errhint("When using timestamptz-version of time_bucket_ng(), 'origin' is "
						 "converted to provided 'timezone'".)));
```

Reply:
Good point, thanks!